serve/mlc_serve/engine/staging_engine.py

```diff
@@ -171,11 +178,25 @@ def step(self) -> InferenceStepResult:
             has_pending_requests=self.has_pending_requests(),
         )
 
+        outputs = list[RequestOutput]()
+
+        with self.requests_to_be_cancelled_lock:
+            if len(self.requests_to_be_cancelled) > 0:
+                for req_id, err_msg in self.requests_to_be_cancelled.items():
+                    outputs.append(
+                        RequestOutput(
+                            req_id,
+                            sequences=[],
+                            error=err_msg,
```

Comment:
By returning a non-None error here, the exception is now raised at https://github.com/octoml/mlc-llm/blob/batch-serving/serve/mlc_serve/engine/async_connector.py#L88-L89

Reply:
Yes as long as we get an exception back we should catch and convert it properly!